devices/Ds18b20/Ds18b20.cs

```diff
@@ -0,0 +1,545 @@
+
+            _oneWireHost.TouchReset();
+
+            if (Address == null)
+            {
+                if (_manyDevicesConnected)
+                {
+                    if (_oneWireHost.FindFirstDevice(false, IsAlarmSearchCommandEnabled))
+                    {
+                        do
+                        {
+                            if (_oneWireHost.SerialNumber[0] == (byte)RomCommands.FamilyCode)
+                            {
+                                _oneWireHost.TouchReset();
+                                Address = new byte[_oneWireHost.SerialNumber.Length];
+                                Array.Copy(_oneWireHost.SerialNumber, Address, _oneWireHost.SerialNumber.Length);
+                                foundDevices++;
+                                allDevices.Add(Address);
+                            }
+                        }
+                        while (_oneWireHost.FindNextDevice(false, IsAlarmSearchCommandEnabled));
+                    }
+
+                    if (foundDevices > 0)
+                    {
+                        AddressNet = new byte[foundDevices][];
+                        int i = 0;
+                        foreach (byte[] device in allDevices)
+                        {
+                            AddressNet[i] = new byte[device.Length];
+                            Array.Copy(device, AddressNet[i], device.Length);
+                            i++;
+                        }
+                    }
+                }
+                else
+                {
+                    if (_oneWireHost.FindFirstDevice(true, IsAlarmSearchCommandEnabled))
+                    {
+                        do
+                        {
+                            if (_oneWireHost.SerialNumber[0] == (byte)RomCommands.FamilyCode)
+                            {
+                                Address = new byte[_oneWireHost.SerialNumber.Length];
+                                Array.Copy(_oneWireHost.SerialNumber, Address, _oneWireHost.SerialNumber.Length);
+                                foundDevices = 1;
+                                break;
+                            }
+                        }
+                        while (_oneWireHost.FindNextDevice(true, IsAlarmSearchCommandEnabled));
+                    }
+                }
+            }
+
+            return foundDevices > 0;
+        }
+
+        /// <summary>
+        /// Reads the temperature. A return value indicates whether the reading succeeded.
+        /// </summary>
+        /// <param name="temperature">
+        /// Contains the measured temperature if the sensor reacted on request.
+        /// Contains -999.99 otherwise.
+        /// </param>
+        /// <returns><code>true</code> if measurement was not skipped, otherwise <code>false</code>.</returns>
+        [Telemetry("Temperature")]
+        public bool TryReadTemperature(out Temperature temperature) => Read(out temperature);
```

Comment:
the read function never return an exception right? It should not because the Try pattern should not!

Reply:
That is correct